### PR TITLE
fixes #877 Additives are always displayed in English while we have tr…

### LINKED ISSUE #877: Additives are always displayed in English while we have translations
https://world.openfoodfacts.org/api/v0/product/737628064502.json

https://world-fr.openfoodfacts.org/additives.json
https://world-ru.openfoodfacts.org/additives.json
https://world.openfoodfacts.org/additives.json

image

### PR DESCRIPTION
**Pull request by Shubham2320**
## Description

On `PreferanceChangeListener` of **PreferencesFragment**, I've called a method `GetAdditives` which uses AsyncTask to get additives in the selected language and stores it in the **greenDao** database so that next time when a user visits **IngredientsProductFragment** he/she can see additives in the selected language.

At present only English and French translation can be seen.

## Related issues and discussion
#877

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java [hunk 1]
```diff
@@ -62,14 +90,80 @@ public boolean onPreferenceClick(Preference preference) {
...
+        }
+
+        @Override
+        protected Boolean doInBackground(Void... arg0) {
+            boolean result = true;
+            Context context = getActivity();
+
+            String additivesFile;
+            switch (LocaleHelper.getLanguage(context)) {
+                case "fr":
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java]: Either doing a very long case system with all the cases, or concatenate languages, and handling languages with non standard codes.

Comment (Shubham2320) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java]: yeah, I thought an approach before to do something like this:
```java
String language = LocaleHelper.getLanguage(SplashActivity.this);
            String additivesFile = "additives_en.json";
            additivesFile.replace("en", language);
```
but I'm unsure about having translation in every language in server at present.

Comment (Shubham2320) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java]: But if the user has changed the language and the selected language doesn't contain the translation so in that case, additives would not be changed to the new language and remain as it is.
It seems quite approachable by the code I've shown.

**Comment by Shubham2320** [comment]
@teolemon I've made the changes please review.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java [hunk 2]
```diff
@@ -62,14 +90,67 @@ public boolean onPreferenceClick(Preference preference) {
...
+            lt.show();
+        }
+
+        @Override
+        protected Boolean doInBackground(Void... arg0) {
+            boolean result = true;
+
+            String language = LocaleHelper.getLanguage(getActivity());
+            String additivesFile = "additives_en.json";
+            additivesFile.replace("en", language);
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java]: Why not replace with these three lines with one?
`String additivesFile = "additives_" + LocaleHelper.getLanguage(getActivity()) + ".json";`

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SplashActivity.java [hunk 3]
```diff
@@ -105,21 +100,15 @@ protected Boolean doInBackground(Void... arg0) {
...
-                    additivesFile = "additives_fr.json";
-                    break;
-                case "en":
-                default:
-                    additivesFile = "additives_en.json";
-                    break;
-            }
+            String language = LocaleHelper.getLanguage(SplashActivity.this);
+            String additivesFile = "additives_en.json";
+            additivesFile.replace("en", language);
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SplashActivity.java]: Same as previous comment.

**Comment by Shubham2320** [comment]
I've made the changes and tested it by switching from English to French in this case the app works fine. I'm not sure about other languages but I've found in API that translations in other languages are also there. It'll take some more time to test it completely :).

**Comment by Karljoones** [comment]
Merged, thanks @Shubham2320 :)

**Comment by Shubham2320** [comment]
Thanks for merging my code :).
However, some more changes would be required in it in future to allow functionality in every language.

**Pull request merged**